enginetest/enginetests.go

```diff
@@ -1736,6 +1736,196 @@ func TestDropColumn(t *testing.T, harness Harness) {
 }
 
+func TestCreateDatabase(t *testing.T, harness Harness) {
+	e := NewEngine(t, harness)
+	ctx := NewContext(harness)
+
+	t.Run("CREATE DATABASE and create table", func(t *testing.T) {
+		TestQuery(t, harness, e,
+			"CREATE DATABASE testdb",
+			[]sql.Row{{sql.OkResult{RowsAffected: 1}}},
+			nil,
+		)
+
+		db, err := e.Catalog.Database("testdb")
+		require.NoError(t, err)
+
+		TestQuery(t, harness, e,
+			"USE testdb",
+			[]sql.Row(nil),
+			nil,
+		)
+
+		require.Equal(t, ctx.GetCurrentDatabase(), "testdb")
+
+		ctx = NewContext(harness)
+		TestQuery(t, harness, e,
+			"CREATE TABLE test (pk int primary key)",
+			[]sql.Row(nil),
+			nil,
+		)
+
+		db, err = e.Catalog.Database("testdb")
+		require.NoError(t, err)
+
+		_, ok, err := db.GetTableInsensitive(ctx, "test")
+
+		require.NoError(t, err)
+		require.True(t, ok)
+	})
+
+	t.Run("CREATE DATABASE IF NOT EXISTS", func(t *testing.T) {
+		TestQuery(t, harness, e,
+			"CREATE DATABASE IF NOT EXISTS testdb2",
+			[]sql.Row{{sql.OkResult{RowsAffected: 1}}},
+			nil,
+		)
+
+		db, err := e.Catalog.Database("testdb2")
+		require.NoError(t, err)
+
+		TestQuery(t, harness, e,
+			"USE testdb2",
+			[]sql.Row(nil),
+			nil,
+		)
+
+		require.Equal(t, ctx.GetCurrentDatabase(), "testdb2")
+
+		ctx = NewContext(harness)
+		TestQuery(t, harness, e,
+			"CREATE TABLE test (pk int primary key)",
+			[]sql.Row(nil),
+			nil,
+		)
+
+		db, err = e.Catalog.Database("testdb2")
+		require.NoError(t, err)
+
+		_, ok, err := db.GetTableInsensitive(ctx, "test")
+
+		require.NoError(t, err)
+		require.True(t, ok)
+	})
+
+	t.Run("CREATE SCHEMA", func(t *testing.T) {
+		TestQuery(t, harness, e,
+			"CREATE SCHEMA testdb3",
+			[]sql.Row{{sql.OkResult{RowsAffected: 1}}},
+			nil,
+		)
+
+		db, err := e.Catalog.Database("testdb3")
+		require.NoError(t, err)
+
+		TestQuery(t, harness, e,
+			"USE testdb3",
+			[]sql.Row(nil),
+			nil,
+		)
+
+		require.Equal(t, ctx.GetCurrentDatabase(), "testdb3")
+
+		ctx = NewContext(harness)
+		TestQuery(t, harness, e,
+			"CREATE TABLE test (pk int primary key)",
+			[]sql.Row(nil),
+			nil,
+		)
+
+		db, err = e.Catalog.Database("testdb3")
+		require.NoError(t, err)
+
+		_, ok, err := db.GetTableInsensitive(ctx, "test")
+
+		require.NoError(t, err)
+		require.True(t, ok)
+	})
+
+	t.Run("CREATE DATABASE error handling", func(t *testing.T) {
+		AssertErr(t, e, harness, "CREATE DATABASE mydb", sql.ErrCannotCreateDatabaseExists)
+
+		AssertWarning(t, e, harness, "CREATE DATABASE IF NOT EXISTS mydb", 1007)
+	})
+}
+
+func TestDropDatabase(t *testing.T, harness Harness) {
+	e := NewEngine(t, harness)
+
+	t.Run("DROP DATABASE correctly works", func(t *testing.T) {
+		TestQuery(t, harness, e,
+			"DROP DATABASE mydb",
+			[]sql.Row{{sql.OkResult{RowsAffected: 1}}},
+			nil,
+		)
+
+		_, err := e.Catalog.Database("mydb")
+		require.Error(t, err)
+
+		// TODO: Deal with handling this error.
+		//AssertErr(t, e, harness, "SHOW TABLES", sql.ErrNoDatabaseSelected)
```

Comment:
@zachmu Are you fine with me leaving this as an issue an moving on? Getting this render correctly seems to be a pain to get right and is not an important feature imo.

Reply:
Yeah I think that's fine for now